microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/HelidonProxyServices.java

```diff
@@ -110,39 +133,65 @@ public boolean supportsClassDefining() {
 
         LOGGER.finest("Defining class " + className + " original class: " + originalClass.getName());
 
+        MethodHandles.Lookup lookup;
+
         try {
-            Module classModule = originalClass.getModule();
-            if (!myModule.canRead(classModule)) {
+            // lookup class name "guessed" from the class name of the proxy
+            // proxy name must contain the $ - if it does not, we just use the originalClass as that is safe
+            int index = className.indexOf('$');
+
+            Class<?> lookupClass;
+            if (index < 0) {
+                LOGGER.finest(() -> "Attempt to define a proxy class without a $ in its name. Class name: " + className + ","
+                                      + " original class name: " + originalClass.getName());
+                lookupClass = originalClass;
+            } else {
+                // I would like to create a private lookup in the same package as the proxied class, so let's do it
+                // use the "extracted" lookup class name, if that fails, use the original class
+                lookupClass = tryLoading(originalClass, className.substring(0, className.indexOf('$')));
```

Comment:
tiny nit: you already stored `index`; just reuse it instead of `className.indexOf('$')` again